src/hotspot/share/runtime/lockStack.hpp

```diff
@@ -35,10 +35,11 @@ class OopClosure;
 class LockStack {
   friend class VMStructs;
 private:
-  static const size_t INITIAL_CAPACITY = 1;
```

Comment:
Uh where did the '1' here come from? I certainly wanted to make this value at least 4. 4 seems to capture a majority of locking depths while not being excessively large. With just 1 or 2 I would expect to blow out the caching advantage with a significant number of workloads. This is probably a left-over from when I wanted to test the growing parts, which I haven't reverted. Please be generous and make that initial size 3 or 4, unless you have objections to that.

Reply:
Done